build/opt.go

```diff
@@ -237,6 +237,11 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt *O
 					opt.Exports[i].Output = func(_ map[string]string) (io.WriteCloser, error) {
 						return w, nil
 					}
+					// if docker is using the containerd snapshotter, prefer to export the image digest
+					// (rather than the image config digest). See https://github.com/moby/moby/issues/45458.
+					if features[dockerutil.OCIImporter] {
```

Comment:
You shouldn't look at the driver features here but the `dapi` inside the `docker.LoadImage`.

Reply:
It seems strange to put that inside `docker.LoadImage()` because that function has a very specific task (i.e., to return a function that will perform the image load). 

The way it's now seems cleaner and addresses the points we discussed yesterday: the code only executes when we know we will load the image, and it reuses the existing Docker client API.

Reply:
Was again confused by this not being driver features but a variable from `dockerutil`. This can be cleaned up in follow-up to avoid creating separate clients for `LoadImage` and `Features` happening sequentially. I already left a comment in [here](https://docker.slack.com/archives/C02JGHEPZJ5/p1745348065164869) that this whole `Features` call is broken an incorrectly cached.

Logically I don't think there is anything wrong with this being in the `LoadImage` caller as what you are asking is the image format of the image loader. But if the `docker info` has already happened and saved to `features` var then should of course avoid it going to waste and doing another call.

Reply:
OK thanks, I'll leave it as is then.